light-rpc/src/main.rs

```diff
@@ -1,4 +1,503 @@
+            meta: Self::Metadata,
+            data: String,
+            config: Option<RpcSendTransactionConfig>,
+        ) -> Result<String> {
+            let RpcSendTransactionConfig {
+                skip_preflight,
+                preflight_commitment,
+                encoding,
+                max_retries,
+                min_context_slot,
+            } = config.unwrap_or_default();
+            let tx_encoding = encoding.unwrap_or(UiTransactionEncoding::Base58);
+            let binary_encoding = tx_encoding.into_binary_encoding().ok_or_else(|| {
+                Error::invalid_params(format!(
+                    "unsupported encoding: {}. Supported encodings: base58, base64",
+                    tx_encoding
+                ))
+            })?;
+            let (wire_transaction, transaction) =
+                decode_and_deserialize::<VersionedTransaction>(data.clone(), binary_encoding)?;
+
+            {
+                let mut lock = meta.signature_status.write().unwrap();
+                lock.insert(transaction.signatures[0].to_string(), None);
+            }
+            meta.tpu_client
+                .send_wire_transaction(wire_transaction.clone());
+            Ok(transaction.signatures[0].to_string())
+        }
+
+        fn get_recent_blockhash(
+            &self,
+            meta: Self::Metadata,
+            commitment: Option<CommitmentConfig>,
+        ) -> Result<RpcResponse<RpcBlockhashFeeCalculator>> {
+            let recent_block_hash = meta.rpc_client.get_latest_blockhash().unwrap();
+            let lamport_per_signature = meta
+                .rpc_client
+                .get_fee_calculator_for_blockhash(&recent_block_hash)
+                .unwrap();
+            let slot = meta.rpc_client.get_slot().unwrap();
+            Ok(RpcResponse {
+                context: RpcResponseContext::new(slot),
+                value: RpcBlockhashFeeCalculator {
+                    blockhash: recent_block_hash.to_string(),
+                    fee_calculator: lamport_per_signature.unwrap(),
+                },
+            })
+        }
+
+        fn confirm_transaction(
+            &self,
+            meta: Self::Metadata,
+            signature_str: String,
+            commitment_cfg: Option<CommitmentConfig>,
+        ) -> Result<RpcResponse<bool>> {
+            let lock = meta.signature_status.read().unwrap();
+            let k_value = lock.get_key_value(&signature_str);
+            let commitment = match commitment_cfg {
+                Some(x) => x.commitment,
+                None => CommitmentLevel::Confirmed,
+            };
+            let confirmed_slot = meta.confirmation_slot.load(Ordering::Relaxed);
+            let finalized_slot = meta.finalized_slot.load(Ordering::Relaxed);
+
+            match k_value {
+                Some(value) => match value.1 {
+                    Some(commitmentForSignature) => {
+                        let slot = if commitment == CommitmentLevel::Finalized {
```

Comment:
the way up to this point is a bit hard to reason. maybe it would help to narrow down the options for `commitment_cfg` very early in the method